src/daemon/inout_route/chat.rs

```diff
@@ -92,15 +97,99 @@ pub fn get_chat(ctx: &DaemonContext, neigh: Fingerprint) -> Vec<(bool, String, S
+pub async fn send_chat_msg(
+    ctx: &DaemonContext,
+    dest: Fingerprint,
+    msg: String,
+) -> anyhow::Result<()> {
     let chats = ctx.get(CHATS);
+    let my_sk = *ctx.get(GLOBAL_IDENTITY);
+    let settlements = ctx.get(SETTLEMENTS);
 
-    if let Some(client) = chats.clients.get(&dest) {
+    if msg.starts_with("!settle ") {
+        let tokens: Vec<&str> = msg.split(' ').collect();
+        let maybe_amount = if tokens.len() == 2 {
+            match tokens[1].parse::<u64>() {
+                Ok(amount) => Some(amount),
+                Err(_) => {
+                    log::warn!("invalid settlement syntax. !settle <amount in micromel>");
+                    None
+                }
+            }
+        } else {
+            None
+        };
+
+        if let Some(amount) = maybe_amount {
+            if let Some(client) = chats.clients.get(&dest) {
+                let proof = SettlementProof::Manual;
+                let req_msg_str = format!("sent you a settlement request for {amount}. Accept with '!accept' or reject with '!reject'.");
+                let req_msg = format!(
+                    "<{}> {}",
+                    my_sk.public().fingerprint().to_string().purple().bold(),
+                    req_msg_str.purple().bold()
+                );
+
+                match client.push_chat(req_msg.clone()).await {
+                    Ok(_) => chats.insert(dest, ChatEntry::new_outgoing(msg)),
+                    Err(e) => log::warn!("error pushing chat: {e}"),
+                };
+
+                let response = client
+                    .start_settlement(SettlementRequest::new(my_sk, amount, proof))
+                    .await;
+                dbg!(&response);
+                let res_msg = match response {
+                    Ok(Some(res)) => {
+                        let descriptor = ctx
+                            .get(RELAY_GRAPH)
+                            .read()
+                            .identity(&dest)
+                            .context(format!("missing neighbor <{dest}> in relay graph"))?;
+                        descriptor
+                            .identity_pk
+                            .verify(res.to_sign().as_bytes(), &res.signature)?;
+
+                        format!("<{dest}> accepted your settlement request for {amount} micromel")
+                            .green()
+                            .bold()
+                    }
+                    Ok(None) => {
+                        format!("<{dest}> rejected your settlement request for {amount} micromel")
+                            .red()
+                            .bold()
+                    }
+                    Err(e) => format!("error sending <{dest}> a settlement request: {e}")
+                        .red()
+                        .bold(),
+                };
+
+                chats.insert(
+                    dest,
+                    ChatEntry::new_incoming(format!("{}", res_msg.bold().blue())),
```

Comment:
After a bit more thought, I think a better design for the chat protocol overall would be to not have the logic here, but rather in the control protocol *client*, and expose separate verbs in the control protocol for manual settlement.

The reason is that the localhost control protocol should be useful for a variety of frontends --- for instance, a GUI frontend that has separate panels for chatting and settling debts.

Thus, stuff like coloring responses, etc, should be done on the frontend. The control protocol server should return UI-agnostic JSON representations and such, not strings to be directly displayed on the GUI.

For now, we can merge it in, since changing this whole approach is something that will touch on other parts of the control protocol too.